broker/lib/fabrik/UnlockResourcePoller.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const pubsub = require('pubsub-js');
+const eventmesh = require('../../../eventmesh');
+const CONST = require('../constants');
+const logger = require('../logger');
+const lockManager = require('./../../../eventmesh').lockManager;
+const errors = require('../errors');
+const config = require('../config');
+const NotFound = errors.NotFound;
+
+class UnlockResourcePoller {
+  static start() {
+    function poller(object, interval) {
+      const lockDetails = JSON.parse(object.spec.options);
+      return eventmesh.apiServerClient.getResource(lockDetails.lockedResourceDetails.resourceGroup, lockDetails.lockedResourceDetails.resourceType, lockDetails.lockedResourceDetails.resourceId)
+        .then((resource) => {
+          const resourceState = resource.body.status.state;
+          logger.debug(`[Unlock Poller] Got resource ${lockDetails.lockedResourceDetails.resourceId} state as `, resourceState);
+          if (resourceState === CONST.APISERVER.RESOURCE_STATE.SUCCEEDED || resourceState === CONST.APISERVER.RESOURCE_STATE.FAILED || resourceState === CONST.APISERVER.RESOURCE_STATE.ERROR) {
+            return lockManager.unlock(object.metadata.name)
+              .then(() => clearInterval(interval));
+          }
+        })
+        .catch(NotFound, err => {
+          logger.info('Resource not found : ', err);
+          return lockManager.unlock(object.metadata.name)
+            .then(() => clearInterval(interval));
+        });
+    }
+    /*
+    Starts poller whenever a lock resource is created.
+    Polling for only backup operations
+    */
+
+    function startPoller(event) {
+      logger.debug('Received Lock Event: ', event);
+      const lockDetails = JSON.parse(event.object.spec.options);
+      if (event.type === CONST.API_SERVER.WATCH_EVENT.ADDED && lockDetails.lockedResourceDetails.resourceGroup === CONST.APISERVER.RESOURCE_GROUPS.BACKUP) {
+        // startPoller(event.object);
+        logger.info('starting unlock resource poller for deployment ', event.object.metadata.name);
+        const interval = setInterval(() => poller(event.object, interval), CONST.UNLOCK_RESOURCE_POLLER_INTERVAL);
```

Comment:
Like i mentioned in the above, its better to convert this to a generic unlocker, which unlocks all types of resources.

Reply:
It can watch on all resources which have completed their operation whose state can be 'Done' and post unlocking it can update it as  'Completed'.